ika/client.go

```diff
@@ -11,6 +11,12 @@ import (
 	"github.com/rs/zerolog"
 )
 
+// IClient defines the methods required for interacting with the Ika network.
+type IClient interface {
```

Comment:
1. Rename this interface to `Client`
2. Rename the struct `Client` to `client`
3. the `NewClient` constructor should return interface, rather than a struct.

Reply:
Done